### PR TITLE
Added random-string-gen package types to repository

### PR DESCRIPTION
**Pull request by MadeByRaymond**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [random-string-gen ] [Run `npm test random-string-gen`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/random-string-gen/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Type definitions for random-string-gen 1.1
+// Project: https://github.com/MadeByRaymond/randomStringGenerator
```
Comment (peterblazejewicz) [inline types/random-string-gen/index.d.ts:2]: what's the project home?
https://github.com/yuhenabc/random#readme

File: types/random-string-gen/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+type Type = "alphanumeric" | "alphabetic" | "numeric" | "hex" | "binary" | "octal" | "ascii-printable";
+type Capitalization = "lowercase" | "uppercase";
+interface GenerateOptions {
+    /**
+     * This is the length of the returned string. Setting this to `0` or a negative
+     * eg: `-1` means an empty string will be returned.
+     *
+     *
+     * [Default : 32]
```
Comment (martin-badin) [inline types/random-string-gen/index.d.ts:14]:
```suggestion
     * @default 32
```

File: types/random-string-gen/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     * @param alphanumeric [0-9 a-z A-Z]
+     * @param alphabetic [a-z A-Z]
+     * @param numeric [0-9]
+     * @param hex [0-9 a-f]
+     * @param binary [01]
+     * @param octal [0-7]
+     * @param ascii-printable [0-7 a-z A-Z !"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~ ]
+     *
+     *
+     * [Default : 'alphanumeric']
```
Comment (martin-badin) [inline types/random-string-gen/index.d.ts:29]:
```suggestion
     * @default 'alphanumeric'
```

File: types/random-string-gen/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    type?: Type;
+    /**
+     * Define a custom character set to use.
+     * This overrides whatever `type` is set.
+     *
+     *
+     * [Default : ' ']
```
Comment (martin-badin) [inline types/random-string-gen/index.d.ts:37]:
```suggestion
     * @default ' '
```

File: types/random-string-gen/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    charset?: string;
+    /**
+     * Define whether the output should be `lowercase` or `uppercase` only.
+     * Ignore this option to return both in the string.
+     *
+     *
+     * [Default : null]
```
Comment (martin-badin) [inline types/random-string-gen/index.d.ts:45]:
```suggestion
     * @default null
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/random-string-gen/index.d.ts (hunk 1 above)
Comment (MadeByRaymond) [reply types/random-string-gen/index.d.ts:2]: https://www.npmjs.com/package/random-string-gen
https://github.com/MadeByRaymond/randomStringGenerator#readme

**Comment by MadeByRaymond** [comment]
Comments addressed @peterblazejewicz @martin-badin

File: types/random-string-gen/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/random-string-gen/index.d.ts:2]: thx, I've read package name wrongly

**Comment by peterblazejewicz** [comment]
Please update project readme, this was the place that mislead me:
https://github.com/MadeByRaymond/randomStringGenerator#readme
it's all the other package

**Comment by MadeByRaymond** [comment]
> Please update project readme, this was the place that mislead me:
> https://github.com/MadeByRaymond/randomStringGenerator#readme
> it's all the other package

This is the repository on github. The package name is different from the repo name this is because the package randomStringGenerator already exists on npm.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@MadeByRaymond thx!

**Comment by MadeByRaymond** [comment]
Ready to merge

**Pull request merged**